internal/app/udevd/main.go

```diff
@@ -10,50 +10,45 @@ import (
 	"fmt"
 	"log"
 	"os"
-	"path"
+	"path/filepath"
 
+	"github.com/mdlayher/kobject"
 	"github.com/pkg/errors"
 	"github.com/talos-systems/talos/internal/app/udevd/internal/drivers/scsi"
-	"github.com/talos-systems/talos/internal/app/udevd/internal/uevent"
 )
 
 func init() {
 	log.SetFlags(log.Lshortfile | log.Ldate | log.Lmicroseconds | log.Ltime)
 }
 
 func watch() error {
-	k, err := uevent.Dial()
+	client, err := kobject.New()
 	if err != nil {
 		return err
 	}
-	// nolint: errcheck
-	defer k.Close()
-
-	for u := range k.Watch() {
-		if u.Error != nil {
-			log.Printf("uevent error: %+v\n", err)
-			continue
+	for {
+		event, err := client.Receive()
+		if err != nil {
+			log.Printf("failed to receive event: %v", err)
 		}
-		if err := handle(u.Message); err != nil {
-			log.Printf("event handler error: %+v\n", err)
+		if err = handle(event); err != nil {
+			log.Printf("%v", err)
 		}
 	}
-
-	return nil
 }
 
 // nolint: gocyclo
-func handle(message *uevent.Message) (err error) {
-	if message.Subsystem == uevent.SubsystemBlock {
+func handle(event *kobject.Event) (err error) {
+	if event.Subsystem == "block" {
```

Comment:
Not: if you invert this if statement and do an early return, you can remove the indentation from this entire block.